secp256k1-zkp-sys/src/zkp.rs

```diff
@@ -476,6 +532,39 @@ impl hash::Hash for PedersenCommitment {
     }
 }
 
+/// A ring signature for the "whitelist" scheme.
+#[repr(C)]
+#[derive(Clone)]
+pub struct WhitelistSignature {
+    /// The number of keys.
+    pub n_keys: size_t,
+    /// The signature in the form of e0 + n_keys s values.
+    pub data: [u8; 32 * (1 + WHITELIST_MAX_N_KEYS)],
+}
+
+impl hash::Hash for WhitelistSignature {
+    fn hash<H: hash::Hasher>(&self, state: &mut H) {
+        self.n_keys.hash(state);
+        self.data[..].hash(state);
+    }
+}
+
+impl PartialEq for WhitelistSignature {
+    fn eq(&self, other: &Self) -> bool {
+        self.n_keys == other.n_keys && self.data[..] == other.data[..]
+    }
+}
+impl Eq for WhitelistSignature {}
+
+impl Default for WhitelistSignature {
+    fn default() -> WhitelistSignature {
+        WhitelistSignature {
+            n_keys: 0,
+            data: [0; 32 * (1 + WHITELIST_MAX_N_KEYS)],
+        }
+    }
+}
```

Comment:
I remember that there was a discussion in the past that these initializers might better be marked `unsafe`. Not sure if that applies here as well.